lib/secretsManager.js

```diff
@@ -1,77 +1,169 @@
-let DUMMY_UUID_GEN = 0
-let DUMMY_STORE = []
-
-/**
- * Find all secrets, applying a selector if provided.
- * When provided, all where clause must be fulfilled (logical AND).
- * By default, returns all secrets
- * @async
- * @param {Object} where - hash of where clauses returned secrets must fulfill with
- * @returns {Array<Object>} list (may be empty) of matching secrets
- */
-export async function find(where = {}) {
-  let secrets = []
-
-  try {
-    const props = Object.keys(where)
-    secrets = DUMMY_STORE.filter(secret =>
-      props.every(prop => secret[prop] === where[prop])
+import { v4 as uuid } from 'uuid'
+import aesjs from 'aes-js'
+import * as SecureStore from 'expo-secure-store'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+
+const encryptionPassphraseStorageKey = 'encryptionPassphrase'
+const secretStorageKey = 'secrets'
+
+export default class SecretsManager {
+  constructor() {
+    this.encryptionPassphrase = null
+  }
+
+  async initialize() {
+    let getEncryptionKey = AsyncStorage.getItem
+    let setEncryptionKey = AsyncStorage.setItem
+
+    if (SecureStore.isAvailableAsync()) {
+      getEncryptionKey = SecureStore.getItemAsync
+      setEncryptionKey = SecureStore.setItemAsync
+    }
```

Comment:
nit on syntax, to avoid mutable variables.
major problem in not awaiting the result of isAvailableAsync

```suggestion
    const [getEncryptionKey, setEncryptionKey] = await SecureStore.isAvailableAsync() 
      ? [SecureStore.getItemAsync, SecureStore.setItemAsync] 
      : [AsyncStorage.getItem, AsyncStorage.setItem]
```

Reply:
Had to double check this one! you're super correct here. I really thought I read this function was synchronous but yes, it isn't. https://docs.expo.io/versions/latest/sdk/securestore/#returns

Reply:
And very nice usage of array destructuring on your suggestion. I'll take it.